src/utils/group.ts

```diff
@@ -20,51 +20,67 @@
  */
 
 import { assert } from "../util";
+import { View, ViewMode } from "../components/editor/View";
 import { File } from "../model";
 
 export default class Group {
-  file: File;
-  files: File[];
-  preview: File;
-  constructor(file: File, preview: File, files: File[]) {
-    this.file = file;
-    this.preview = preview;
-    this.files = files;
+  currentView: View;
+  views: View[];
+  preview: View;
+
+  constructor(view: View, views: View[] = []) {
+    this.currentView = view;
+    this.views = views;
   }
-  open(file: File, shouldPreview: boolean = true) {
-    const files = this.files;
-    const index = files.indexOf(file);
+  open(view: View, shouldPreview = true) {
+    const views = this.views;
+    const index = views.indexOf(view);
+
     if (index >= 0) {
-      // Switch to file if it's already open.
-      this.file = file;
+      // Switch to the view if it's already open.
+      this.currentView = view;
       if (!shouldPreview) {
         this.preview = null;
       }
-      return;
+
+      return ;
```

Comment:
nit: can we remove line break, I don't think this change is needed. (it rather belong after the `if`)